solidity/test/token_stake/TestStakingPortBacker.js

```diff
@@ -0,0 +1,635 @@
+      expect(await newTokenStaking.beneficiaryOf(operatorThree)).to.equal(beneficiaryThree)
+      expect(await newTokenStaking.authorizerOf(operatorThree)).to.equal(authorizerThree)
+      expect(await newTokenStaking.balanceOf(operatorThree)).to.eq.BN(delegatedAmount)
+      expect(await newTokenStaking.ownerOf(operatorThree)).to.equal(stakingPortBacker.address)
+    })
+
+    it("allows to copy stake only one time", async () => {
+      await stakingPortBacker.copyStake(operatorOne, {from: tokenOwner})
+      await stakingPortBacker.copyStake(operatorTwo, {from: grantee})
+      await stakingPortBacker.copyStake(operatorThree, {from: managedGrantee})
+
+      await expectRevert(
+        stakingPortBacker.copyStake(operatorOne, {from: tokenOwner}),
+        "Stake already copied"
+      )
+      await expectRevert(
+        stakingPortBacker.copyStake(operatorTwo, {from: grantee}),
+        "Stake already copied"
+      )
+      await expectRevert(
+        stakingPortBacker.copyStake(operatorThree, {from: managedGrantee}),
+        "Stake already copied"
+      )
+    })
+
+    it("emits an event", async () => {
+      let receipt = await stakingPortBacker.copyStake(operatorOne, {from: tokenOwner})
+      await expectEvent(receipt, "StakeCopied", {
+        owner: tokenOwner,
+        operator: operatorOne,
+        value: delegatedAmount          
+      })
+    })
+  })
+
+  describe("repaying backed balances", async () => {
+    beforeEach(async () => {
+      await stakingPortBacker.allowOperators(
+        [operatorOne, operatorTwo, operatorThree], {from: owner}
+      )
+
+      await stakingPortBacker.copyStake(operatorOne, {from: tokenOwner})
+      await stakingPortBacker.copyStake(operatorTwo, {from: grantee})
+      await stakingPortBacker.copyStake(operatorThree, {from: managedGrantee})
+    })
+
+    it("fails for unknown token", async () => {
+      let anotherToken = await KeepToken.new({from: deployer})
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await expectRevert(
+        anotherToken.approveAndCall(
+          stakingPortBacker.address, delegatedAmount, data, {from: tokenOwner}
+        ),
+        "Not a KEEP token"
+      )
+    })
+
+    it("fails for less tokens than backed", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount.subn(1), data, {from: tokenOwner}
+        ),
+        "Unexpected amount"
+      )  
+    })
+
+    it("fails for more tokens than backed", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount.addn(1), data, {from: tokenOwner}
+        ),
+        "Unexpected amount"
+      )  
+    })
+
+    it("fails for operator with no tokens backed", async () => {
```

Comment:
Can we add a case where an operator's token balance is less then the stake?

Reply:
Sure, good idea! Added in 67e6349.